### PR TITLE
Add exception handling for slack client

### PR DESCRIPTION
**Pull request by shekharkhedekar**
https://github.com/slackapi/python-slackclient/wiki/Migrating-to-2.x#error-handling-changes

### CODE REVIEW

**Comment by shekharkhedekar** [comment]
:eyes: @ryan-lane

File: omnibot/processor.py [hunk 1]
```diff
@@ -311,23 +311,27 @@ def _handle_post_message(message, kwargs):
     except json.decoder.JSONDecodeError:
         logger.exception('JSON decode failure when parsing {}'.format(kwargs))
         return
-    logger.debug(ret, extra=message.event_trace)
-    if not ret['ok']:
+    except:
```
Comment (ryan-lane) [inline omnibot/processor.py]: bare excepts also catch signals. you should catch SlackClientError as it's their generic exception

File: omnibot/processor.py [hunk 2]
```diff
@@ -311,23 +311,27 @@ def _handle_post_message(message, kwargs):
...
+            client_type='oauth_bot'
+        ).api_call(
+            action,
+            json=kwargs
+        )
+        logger.debug(ret)
+    except Exception as e:
+        logger.exception('Error in handle action', kv={'error': str(e)})
+        # TODO (shekharkhedekar) validate errors
+        if str(e) == 'missing_scope':
```
Comment (ryan-lane) [inline omnibot/processor.py]: I think you need `e.response.get('error')`

File: omnibot/routes/api.py [hunk 3]
```diff
@@ -641,31 +641,34 @@ def _perform_action(bot, data):
...
+
+            # Remove the file
+            os.remove(file.name)
+        else:
+            ret = slack.client(bot, client_type='oauth_bot').api_call(
+                action,
+                json=kwargs
+            )
+        logger.debug(ret)
+    except Exception as e:
```
Comment (ryan-lane) [inline omnibot/routes/api.py]: If you capture the entire block in an exception like this, you may end up leaking file handles.

I also didn't notice this before, but for files, you should use with:

```
with open(filename, 'wb') as f:
```

File: omnibot/routes/api.py [hunk 4]
```diff
@@ -641,31 +641,34 @@ def _perform_action(bot, data):
...
+        logger.exception('Error uploading file', kv={'error': str(e)})
+        # TODO (shekharkhedekar) validate errors
+        if str(e) in ['missing_scope', 'not_allowed_token_type']:
```
Comment (ryan-lane) [inline omnibot/routes/api.py]: same as above, but you want `e.response.get('error')`

File: omnibot/services/slack/__init__.py [hunk 5]
```diff
@@ -417,49 +418,51 @@ def get_channel(bot, channel):
...
+        try:
+            group_data = client(bot, client_type='oauth_bot').api_call(
+                'groups.info',
+                json={
+                    'channel': channel,
+                }
+            )
+            update_group(bot, group_data['group'])
+            return group_data['group']
+        except:
```
Comment (ryan-lane) [inline omnibot/services/slack/__init__.py]: I think for this you can do a pass in the except and not indent the rest

File: omnibot/processor.py (hunk 1 above)
Comment (shekharkhedekar) [reply omnibot/processor.py]: updated

File: omnibot/processor.py (hunk 2 above)
Comment (shekharkhedekar) [reply omnibot/processor.py]: updated

File: omnibot/routes/api.py (hunk 3 above)
Comment (shekharkhedekar) [reply omnibot/routes/api.py]: good point, updated

File: omnibot/routes/api.py (hunk 4 above)
Comment (shekharkhedekar) [reply omnibot/routes/api.py]: updated

**Pull request merged**